src/vs/editor/common/config/commonEditorConfig.ts

```diff
@@ -959,6 +959,11 @@ const editorConfiguration: IConfigurationNode = {
 			'default': EDITOR_DEFAULTS.contribInfo.showFoldingControls,
 			'description': nls.localize('showFoldingControls', "Controls whether the fold controls on the gutter are automatically hidden.")
 		},
+		'editor.foldingFromEnd': {
```

Comment:
Let's make it an enum, something like:  `editor.foldingControls: "classic" | "top-buttom"

Reply:
That makes sense -- more future proof!  I'll push this up when addressing the other feedback.